numpy/core/src/umath/logical_gufuncs.c.src

```diff
@@ -0,0 +1,302 @@
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+#include "Python.h"
+#include "math.h"
+#include "numpy/ndarraytypes.h"
+#include "numpy/ufuncobject.h"
+#include "numpy/npy_3kcompat.h"
+#include "logical_gufuncs.h"
+
+
+// define the basic real version similar to the macro for complex numbers
+#define REQ(a,b) (a==b)
+#define RNE(a,b) (a!=b)
+#define RLT(a,b) (a<b)
+#define RLE(a,b) (a<=b)
+#define RGT(a,b) (a>b)
+#define RGE(a,b) (a>=b)
+
+/* -------------------------------------------------------------------------- */
+/* define the inner loops  */
+
+#define all_INNER_LOOP(OP, TYPE)                \
+    *((npy_bool *)c_n) = NPY_TRUE;              \
+    for (i = 0; i < I; i++){                    \
+        a = *(TYPE *)a_i;                       \
+        b = *(TYPE *)b_i;                       \
+        if (OP(a,b)){                           \
+            a_i += a_I;                         \
+            b_i += b_I;                         \
+        } else {                                \
+            *((npy_bool *)c_n) = NPY_FALSE;     \
+            break;                              \
+        }                                       \
+    }
+
+#define any_INNER_LOOP(OP, TYPE)                \
+    *((npy_bool *)c_n) = NPY_FALSE;             \
+    for (i = 0; i < I; i++){                    \
+        a = *(TYPE *)a_i;                       \
+        b = *(TYPE *)b_i;                       \
+        if (OP(a,b)){                           \
+            *((npy_bool *)c_n) = NPY_TRUE;      \
+            break;                              \
+        } else {                                \
+            a_i += a_I;                         \
+            b_i += b_I;                         \
+        }                                       \
+    }
+
+/* -------------------------------------------------------------------------- */
+/* create the family of functions using a template  */
+
+/**begin repeat
+* #TYPE = byte,ubyte,short,ushort,int,uint,long,ulong,longlong,ulonglong,float,double,longdouble,cfloat,cdouble,clongdouble#
```

Comment:
The long line should be broken. Indeed, all the long lines should be < 80 characters.